File: fix_article.js
const fs = require('fs');
const path = 'c:/Users/manyc/Falaw/html/Falaw/falaw.com.br/blog/articles-data.js';
let src = fs.readFileSync(path, 'utf8');

// Evaluate to get FALAW_ARTICLES
eval(src);
const a = FALAW_ARTICLES.find(x => x.id.startsWith('stf-are-1532603-sobresto'));
const oldContent = a.content;
console.log('len:', oldContent.length, 'tail:', JSON.stringify(oldContent.slice(-20)));

// Fix: remove the extra } — current tail is "]}}]}" should be "]}]}"
const newContent = oldContent.slice(0, -4) + ']}';
console.log('new tail:', JSON.stringify(newContent.slice(-20)));

try {
  const p = JSON.parse(newContent);
  console.log('Parse OK! sections:', p.sections.length);
} catch(e) {
  console.log('Still broken:', e.message); process.exit(1);
}

// Replace in source file — the content field value is stored as a JSON string in the JS file
// We need to find the exact escaped version in the source
const oldEscaped = JSON.stringify(oldContent).slice(1,-1); // remove outer quotes
const newEscaped = JSON.stringify(newContent).slice(1,-1);
console.log('old escaped tail:', oldEscaped.slice(-20));
console.log('new escaped tail:', newEscaped.slice(-20));

if (!src.includes(oldEscaped)) {
  console.log('Could not find old escaped content in source!'); process.exit(1);
}

const newSrc = src.replace(oldEscaped, newEscaped);
fs.writeFileSync(path, newSrc, 'utf8');
console.log('File updated.');

File: fix_article.py
path = r'c:\Users\manyc\Falaw\html\Falaw\falaw.com.br\blog\articles-data.js'
with open(path, encoding='utf-8') as f:
    data = f.read()

# The content currently ends with: ...]}}]}  (extra })
# Should end with:                  ...]}}    wait - let me think:
# paragraphs: [..., "last sentence."]}  -> ] closes paragraphs, } closes section
# Then we need ] to close sections array, } to close root = so  ]}
# Full: ...last sentence."]}]}
# But currently: ...last sentence."]}}`  — with extra }
# In file source the problematic part is:
old = '19/06/2026."]}]}'  # 2 extra chars vs needed
# Actually from node output, content ends with: 026."]}}`  

# Let me just read the actual bytes:
idx = data.find('stf-are-1532603-sobresto')
content_start = data.find('"content"', idx)
# find the content string value
cs = data.find('"', content_start + len('"content"') + 2) + 1  # skip ": "
# now find the end of the content string (closing " not preceded by \)
import re
m = re.search(r'(?<!\\)"', data[cs:])
ce = cs + m.start()
content_val = data[cs:ce]
print("Content length:", len(content_val))
print("Content tail:", repr(content_val[-30:]))

# Try parsing
import json
try:
    parsed = json.loads(content_val)
    print("OK! sections:", len(parsed['sections']))
except json.JSONDecodeError as e:
    print("Error:", e)
    # Show chars around error position
    pos = e.pos
    print("Around error:", repr(content_val[pos-10:pos+10]))
